File: BOJ/BOJ_11723.java
import java.io.*;
import java.util.*;

public class BOJ_11723 {
	static HashSet<Integer> chkSet = new HashSet<>();

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		int M = Integer.parseInt(br.readLine());

		for (int i = 0; i < M; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine(), " ");
			String command = st.nextToken();
			if (st.hasMoreTokens()) {
				int x = Integer.parseInt(st.nextToken());
				switch (command) {
				case "add":
					chkSet.add(x);
					break;
				case "remove":
					chkSet.remove(x);
					break;
				case "check":
					if (chkSet.contains(x)) {
						sb.append(1).append("\n");

					} else {
						sb.append(0).append("\n");
					}
					break;
				case "toggle":
					if (chkSet.contains(x)) {
						chkSet.remove(x);
					} else {
						chkSet.add(x);
					}
					break;
				}
			} else {
				switch (command) {
				case "all":
					for(int j=1;j<=20;j++) {
						chkSet.add(j);
					}
					break;
				case "empty":
					chkSet.clear();
					break;
				}
			}
		}
		System.out.println(sb);
	} // end of main
} // end of class

File: 백준/BOJ_11723.py
"""
1. 아이디어
    -add
        set.add(x)
    -empty
        set.clear()
    -all
        continue
    if x in set:
        if check: print(1)
        elif toggle: set-={x}
    else:
        if check:print(0)
        elif toggle: set.add(x)
"""
import sys
input = sys.stdin.readline

m = int(input())
s = set()
for _ in range(m):
    command = input().split()
    if command[0] == "all": 
        for i in range(1,21):
            s.add(i)
    elif command[0] == "empty":s.clear()
    else:
        x=int(command[1])
        if command[0]=="add": s.add(x)
        elif command[0]=="remove":s-={x}
        else:
            if x in s:
                if command[0] == "check": print(1)
                else: s-={x}
            else:
                if command[0] == "check": print(0)
                else: s.add(x)
